src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -169,10 +171,11 @@ public void visitToken(DetailAST ast) {
      * @param ast root of given subtree
      * @param useBreak should we consider break as terminator
      * @param useContinue should we consider continue as terminator
+     * @param labels label names
      * @return true if the subtree is terminated.
      */
     private boolean isTerminated(final DetailAST ast, boolean useBreak,
-                                 boolean useContinue) {
+                                 boolean useContinue, Set<String> labels) {
```

Comment:
Each Set of labels is only for a given "level" of switch, right?

Example of my understanding:
```
        top: switch (a) { // 'top' is in one Set of labels
            case 1:
                inner: switch (b) { // 'inner' is in a different Set of labels
                    case 10:
                        break inner;                     
                }
            case 2: // Fall through
                return 2;
        }
```

If my understanding is correct, can we somehow make this obvious? Perhaps improve the Javadoc, and rename this Set to something that makes this obvious?

Reply:
> Each Set of labels is only for a given "level" of switch, right?
> 
> Example of my understanding:
> 
> ```
>         top: switch (a) { // 'top' is in one Set of labels
>             case 1:
>                 inner: switch (b) { // 'inner' is in a different Set of labels
>                     case 10:
>                         break inner;                     
>                 }
>             case 2: // Fall through
>                 return 2;
>         }
> ```
> 
> If my understanding is correct, can we somehow make this obvious? Perhaps improve the Javadoc, and rename this Set to something that makes this obvious?

Yes, you are right. I have made changes to the the name.